### DISCUSSION TITLE
More help on the DT responsiveness extension

### DISCUSSION THREAD

**Discussion by pycolle** [l-lin/angular-datatables#1561] [Q&A]
Hello, 
always super happy to work with this table, I have few question on he DT responsiveness extension ....

**_QUESTION 1_**

Do I understand correctly that it allow to eventually display multiple -- kind of -- sub-rows below one given row when clicking on its + button 
My interrest here is the following : I have some references and for each of them potentially from 1 to N versions with some rules.
By default I will display the most appropriate version but it could be nice if by clicking on this + button it could open the space below the concerned row and show me the remaining N-1 version that potentially could exist ...

**_QUESTION 2_**

I tried to put the code in place exactly as it was written in the example
but I got an error : DataTables warning: table id=DataTables_Table_3 - Ajax error. For more information about this error, please see http://datatables.net/tn/7.

In fact my table is taking its data from a small observable service .. if you remember I put a post and you helped me indicating me that in this case I should put a dtTrigger.next() in the result of the subscribe method so that the table load the data ... 

so my HTML code is this one : 

<table datatable class="row-border hover" [dtOptions]="dtOptions" [dtTrigger]="dtTrigger">
  <thead>
    <tr>
      <th>ID</th>
      <th>First name</th>
      <th>Last name</th>
    </tr>
  </thead>
  <tbody>
    <tr *ngFor="let item of m_DataInPage04 ; let i = index">
      <td>{{i}}</td>
      <td>{{item._FName}}</td>
      <td>{{item._LName}}</td>
    </tr>
  </tbody>
</table>

is it because my m_DataInPage04 structure has not exactly the same name as what you are expecting but instead _FName and _LName ? 

Thanks a lot for yoru coming help

**Comment by shanmukhateja** [comment]
Hi,

 **Question 1**

I don't know how the responsive plugin works under-the-hood. I suspect it adds/removes DOM nodes acc. to the user's viewport (desktop, tablet or mobile). You will need to reach out to DataTables.net forums for that information.

Regarding your feature implementation, I do not fully understand it. Did you mean that whenever a user clicks on a + `<button>` a new row would need to be added after the currently "used" row? This would require using the DataTables.net APIs to dynamically add data and draw the changes to DOM.

If this is your requirement, I may have an alternative for you. Using the TemplateRef feature in recent versions of this plugin,there's a way to inject any Angular component into a `<td>` element in the table at _runtime_ with support for listening to its events and acting on them.  Check the documentation and see if that could help?

**Question 2**

I will need you to provide a small reproducible repo or StackBlitz link to investigate. It could be anything from an incorrect Ajax request or your server failing to process the said request.

> is it because my m_DataInPage04 structure has not exactly the same name as what you are expecting but instead _FName and _LName ?

You can test this yourself. Just use `{{item | json}}` to print out all fields available inside `item`.

**Reply by pycolle** [reply]
Hello shanmukhateja
I will come back to you ASAP on the second problem. Let me first try to solve it by myself ... 
In case by any miracle I succeed, I will in any case come back here to explain how I did and what was wrong in case it could help someone ..
In case after some trials I start to be deseperate I will come back to you ... but as my code has now a back-end part to retreive data from an ORACLE datbase ... it's not so easy to upload a part of it somewhere
